lib/simple_form/components/errors.rb

```diff
@@ -11,7 +11,7 @@ def full_error(wrapper_options = nil)
       end
 
       def has_errors?
-        object && object.respond_to?(:errors) && errors.present?
+        form_with_object_has_errors? || form_without_object_has_errors?
```

Comment:
Why do we need to check if the object is nil at the second condition?, just check `has_custom_errors?` would be enough, is it?

Reply:
I have to check if the object is `nil` to respect other logic based in forms using an object and if there is not error on the attribute. If the `object.nil?` is removed from the second condition then we have two test failing.

Test 1: https://github.com/victorperez/simple_form/blob/master/test/form_builder/error_test.rb#L221
Test 2: https://github.com/victorperez/simple_form/blob/master/test/form_builder/error_test.rb#L274

Reply:
So we could rename this method or just use the conditions here, something like this:

```ruby
object_with_errors? || object.nil? && has_custom_error?
```

Reply:
Done, I have amended my commit changed the conditions to what you propose 👍